### PR TITLE
Rename to stack-gcp-sample

### PR DESCRIPTION
**Pull request by hasheddan**
Opening this in draft to make sure everyone feels comfortable with this fairly cumbersome name change. After sign-off I will move forward with other minimal stacks and also get this repo officially renamed :+1: 

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### CODE REVIEW

File: .registry/app.yaml
Comment (prasek) [inline .registry/app.yaml]:
```suggestion
```

File: .registry/resources/gcp.resourcepacks.crossplane.io_samplek8sandmanagedservices.crd.yaml
Comment (prasek) [inline .registry/resources/gcp.resourcepacks.crossplane.io_samplek8sandmanagedservices.crd.yaml]:
```suggestion
  name: samplek8sandmanagedservices.gcp.stacks.crossplane.io
```

**Review by prasek — COMMENTED** [review]
LGTM overall. Dropped a couple comments. Thanks!

**Review by suskin — COMMENTED** [review]
It seems like a lot of words to say when referring to this specific stack. Do we have a short form yet, or a category term? It seems like it'll be awkward to say "Okay, publishing a new GCP Sample Kubernetes and Managed Services Stack now". Maybe we're thinking "GCP Sample" stack as the short form? Or, I suppose we could still use "Minimal GCP" in conversation.

**Comment by rathpc** [comment]
I am a bit weary on this big hairy name...descriptive names for the title and kind makes sense, on board there, but does that also mean the repo name itself has to match? Seems like we could get creative with the naming there to keep those short and sweet

File: .registry/app.yaml
Comment (rathpc) [inline .registry/app.yaml]: Are we sure we don't want to keep any reference to GCP here? Assuming we follow this pattern this name would "collide" with the other sample stacks, no?

File: .registry/resources/samplek8sandmanagedservices.ui-schema.yaml
Comment (rathpc) [inline .registry/resources/samplek8sandmanagedservices.ui-schema.yaml]: Should probably bump the patch version of the ui-schema if we are changing this

File: README.md
Comment (rathpc) [inline README.md]: Typo here. Should say `provider-gcp`

**Review by rathpc — CHANGES_REQUESTED** [review]

File: .registry/app.yaml
Comment (prasek) [reply .registry/app.yaml]: right, how about “GCP Sample Stack for Kubernetes and Managed Services”

File: README.md
Comment (prasek) [reply README.md]: also maybe bump to v0.7.0 here and definitely v0.7.1 for AWS when we get to that stack.

File: README.md
Comment (rathpc) [inline README.md]: I'd argue to keep the ordering of "sample" and "gcp" the same as the other
```suggestion
kubectl crossplane stack install --cluster -n crossplane-system 'crossplane/stack-gcp-sample-k8s-and-managed-services:<version>' gcp-sample
```

File: README.md
Comment (rathpc) [inline README.md]:
```suggestion
  name: "gcp-sample"
```

File: .registry/behavior.yaml
Comment (suskin) [inline .registry/behavior.yaml:5]: What are your thoughts on putting `samples` in the group so that all of the sample stacks roll into the same group at that level? Example: `gcp.samples.stacks.crossplane.io`

**Review by suskin — APPROVED** [review]
What testing has been done? : )

I had a question, but it looks good to me otherwise. Approving in case we don't make any more changes and are already confident in the correctness of the change. :squid:

File: .registry/behavior.yaml
Comment (rathpc) [reply .registry/behavior.yaml:5]: i like this idea

Comment (hasheddan) [reply .registry/behavior.yaml:5]: That seems fine to me, I could also see wanting it to roll-up to `gcp` before `samples` though

Comment (hasheddan) [reply .registry/behavior.yaml:5]: @suskin @rathpc what you you call the `Kind` in this case? It seems like `gcpsample.gcp.samples.stack.crossplane.io` is pretty repetitive

Comment (rathpc) [reply .registry/behavior.yaml:5]: i think due to that, `gcpsamples.stacks.crossplane.io` is probably fine

File: .registry/resources/gcpsample.resource.yaml
Comment (rathpc) [inline .registry/resources/gcpsample.resource.yaml]:
```suggestion
id: gcpsample
```

Comment (hasheddan) [reply .registry/resources/gcpsample.resource.yaml]: Darn it I missed another one 😅

File: .registry/behavior.yaml
Comment (rathpc) [reply .registry/behavior.yaml:5]: i think we can just leave this as is for now

**Review by rathpc — APPROVED** [review]
one small fix otherwise looks good so I will approve assuming that gets fixed

**Review by jbw976 — COMMENTED** [review]
LGTM after https://github.com/crossplane/stack-minimal-gcp/pull/38/files#r401942396 is resolved

**Comment by hasheddan** [comment]
@jbw976 fixed! I would like to do a quick run-through to make sure everything is working before merge though :+1:

**Comment by jbw976** [comment]
OK great @hasheddan, please @ mention me when you have verified with testing :)

File: .registry/app.yaml
Comment (muvaf) [inline .registry/app.yaml]: Should this be `GCP Sample Stack` or do we want to keep `Environment` here and in other files?

Comment (hasheddan) [reply .registry/app.yaml]: Not sure, I didn't change it because we were using "Environment" even though this is a "Stack" already. I think it was because that is kind of the purpose of this stack, but I would be happy to update 👍 @prasek what do you think?

Comment (prasek) [reply .registry/app.yaml]: I would use "GCP Sample Stack" like we do for "GCP Provider" even though that's a provider already.

**Comment by hasheddan** [comment]
@jbw76 this is ready to go :+1: I tested by installing Crossplane `v0.10.0-rc.26.g18fe287` and `stack-gcp` from `master`. I then used the following configurations to use my local image in a Kind cluster:

```yaml
apiVersion: stacks.crossplane.io/v1alpha1
kind: ClusterStackInstall
metadata:
  name: "gcp-sample"
  namespace: crossplane-system
spec:
  package: "stack-rename:local" # manually loaded this into Kind cluster
  imagePullPolicy: IfNotPresent
```

```yaml
apiVersion: gcp.stacks.crossplane.io/v1alpha1
kind: GCPSample
metadata:
  name: gcp-test-3
spec:
  region: us-west2
  projectID: crossplane-playground
  credentialsSecretRef:
    name: gcp-account-creds
    namespace: crossplane-system
    key: credentials
```

**Review by prasek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
